### DISCUSSION TITLE
How to update the data in ACAD_TABLE please?

### DISCUSSION THREAD

**Discussion by xbdr419** [mozman/ezdxf#955] [Q&A]
How to update the data in ACAD_TABLE please?

**Comment by mozman** [comment]
The `ACAD_TABLE` entity is just preserved, you can not edit or query the content.

**Reply by nquinterno** [reply]
Do you know if the query of the ACAD_TABLE entity  content  will be supported in future releases?

**Reply by mozman** [reply]
Yes, I can answer that: 

No, the `ACAD_TABLE` object is far too complicated and not documented enough to implement.  I know that from my experiences with the `MLEADER` object.

**Reply by nquinterno** [reply]
Thanks for the reply. Do you know if, as an alternative, Excel tables inserted as OLE objects can be read?

**Reply by mozman** [reply]
No, OLE and OL2 entities are also just preserved.

**Comment by mozman** [comment]
You have access to the `DXFTagStorge` entity which stores the tags of unsupported entities:

This script reads the cell values from a simple table:

![image](https://github.com/mozman/ezdxf/assets/4230966/654734f8-0c74-441f-9af3-bbbd5e268e9d)

```python
import ezdxf

def main():
    doc = ezdxf.readfile(r"C:\Source\ezdxf.git\examples_dxf\acad_table_simple.dxf")
    msp = doc.modelspace()
    table = msp.query("ACAD_TABLE").first
    acdb_table = table.xtags.get_subclass("AcDbTable")
    values = [tag.value for tag in acdb_table.find_all(302)]
    print("Table Values:")
    print(values)


if __name__ == "__main__":
    main()
```

output:

```
Table Values:
['Title', '', '', '1', '2', '3', '4', '5', '6']
```

[Here](https://help.autodesk.com/view/OARX/2018/ENU/?guid=GUID-D8CCD2F0-18A3-42BB-A64D-539114A07DA0) is the DXF reference of `ACAD_TABLE` - have fun!

**Reply by mozman** [reply]
I added a helper function to `ezdxf` for that task:

```Python
import ezdxf
from ezdxf.entities.acad_table import read_acad_table_content


def main():
    doc = ezdxf.readfile(r"C:\Source\ezdxf.git\examples_dxf\acad_table_simple.dxf")
    msp = doc.modelspace()
    table = msp.query("ACAD_TABLE").first
    values = read_acad_table_content(table)
    print("Table Values:")
    print(values)

if __name__ == "__main__":
    main()

```
output as list of rows:
```
Table Values:
[['Title', '', ''], ['1', '2', '3'], ['4', '5', '6']]
```

**Reply by nquinterno** [reply]
Thanks for the answer, I was able to read the values ​​of the cells of the acad_table entity. Now it's time to play with them
